jaxrs/WFLY-8136_EAP7-1228_JAX-RS_annotations_in_CDI_Stereotype.adoc

```diff
@@ -0,0 +1,86 @@
+= WFLY-8136 / EAP7-1228 Allow JAX-RS annotations in CDI @Stereotype
+:author:            Martin Stefanko
+:email:             mstefank@redhat.com
+:toc:               left
+:icons:             font
+:idprefix:
+:idseparator:       -
+
+== Overview
+
+This feature provides support for JAX-RS annotations to be defined as a part of 
+CDI `@Stereotype` definitions. This will allow JAX-RS annotations like `@Produces` 
+or `@Consumes` to be reused by several JAX-RS resources or providers.
+
+Example:
+
+[java]
+---- 
+@Stereotype
+@Produces(MediaType.APPLICATION_JSON)
+@Target(ElementType.TYPE)
+@Retention(RetentionPolicy.RUNTIME)
+public @interface RestService {
+}
+
+@Path("/some")
+@RestService
+public class SomeRestService {
+    @GET
+    @Path("/")
+    public Response getSome() {
+        return Response.ok().entity("{\"x\" : 42, \"y\" : \"foo\"}").build();
+    }
+}
+----
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.jboss.org/browse/EAP7-1228[EAP7-1228]
+* https://issues.jboss.org/browse/WFLY-8136[WFLY-8136]
+
+=== Related Issues
+
+* https://issues.jboss.org/browse/RESTEASY-1821[RESTEASY-1821]
+
+=== Dev Contacts
+
+* mailto:mstefank@redhat.com[Martin Stefanko]
+* mailto:asoldano@redhat.com[Alessio Soldano]
+* mailto:rsigal@redhat.com[Ron Sigal]
+
+=== QE Contacts
+
+* TBD
+
+=== Testing By
+// Put an x in the relevant field to indicate if testing will be done by Engineering or QE. 
+// Discuss with QE during the Kickoff state to decide this
+[x] Engineering
+
+[ ] QE
+
+=== Affected Projects or Components
+
+* RESTEasy
+
+=== Other Interested Projects
+
+== Requirements
+
+* RESTEasy will be able to process JAX-RS annotations in CDI Stereotype when run in 
```

Comment:
Is CDI environment always present in WF? Or is it possible that CDI environment won't be available in WF deployment?

Reply:
@marekkopecky it wildfy it should be always present (at least up to my knowledge). However RESTEAsy can be used in environments/containers where CDI is not available.